apiserver/facades/agent/uniter/uniter.go

```diff
@@ -1987,21 +1991,26 @@ func (u *UniterAPI) NetworkInfo(args params.NetworkInfoParams) (params.NetworkIn
 		networkInfos = machine.GetNetworkInfoForSpaces(spaces)
 	} else {
 		// For CAAS units, we build up a minimal result struct
-		// based on the default space and unit private address,
-		// ie the address of the container.
-		addr, err := unit.PrivateAddress()
+		// based on the default space and unit public/private addresses,
+		// ie the addresses of the CAAS service.
+		addr, err := unit.AllAddresses()
 		if err != nil {
 			return params.NetworkInfoResults{}, err
 		}
+		network.SortAddresses(addr)
+
+		// The binding address is that of the pod, the machine local address.
+		// The default ingress addresses are all known unit addresses.
+		var interfaceAddr []network.InterfaceAddress
+		for _, a := range addr {
+			if a.Scope == network.ScopeMachineLocal {
+				interfaceAddr = append(interfaceAddr, network.InterfaceAddress{Address: a.Value})
+			}
```

Comment:
This will add a.Value twice. Is it intended or did you need a "continue" in the if-?

Reply:
It adds the address to both the interfaceAddr slice (used for bind addresses later) and the default ingress addresses. The default addresses are only used if the result ingress addresses are empty. For IAAS models, bind addresses populate the ingress addresses. So we either end up with some ingress addresses or none. In the none case, we use the defaults.

Reply:
tried to clarify comment